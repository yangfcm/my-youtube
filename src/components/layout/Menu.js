import React from "react";
import { NavLink } from "react-router-dom";

const Menu = ({ menuItems = [] }) => {
  return (
    <div>
      <nav className="nav nav-tabs nav-fill">
        {menuItems.map(item => {
          return (
            <NavLink
              key={item.name}
              to={item.link}
              exact={item.exact}
              className="nav-item nav-link"
              activeClassName="active"
            >
              {item.name}
            </NavLink>
          );
        })}
      </nav>
    </div>
  );
};

export default Menu;
