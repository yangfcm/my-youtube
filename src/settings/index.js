import axios from "axios";

export const apiBaseUrl = "https://www.googleapis.com/youtube/v3";

export const defAxios = axios.create({
  baseURL: apiBaseUrl,
  timeout: 10000,
});

export const defaultName = "Y Fan";
export const mainMenuItems = [
  { link: "/", exact: true, name: "Recommend", icon: "thumbs-up" },
  { link: "/channel", exact: false, name: "Channel", icon: "user-plus" },
  { link: "/playlist", exact: false, name: "Play List", icon: "list" },
];

export const maxResults = 12;
