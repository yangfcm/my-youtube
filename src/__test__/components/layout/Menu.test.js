import React from "react";
import { shallow } from "enzyme";
import { NavLink } from "react-router-dom";
import Menu from "components/layout/Menu";
import { mainMenuItems } from "settings";

describe("Test Menu component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Menu menuItems={mainMenuItems} />);
  });

  it("should render component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have correct number of links", () => {
    expect(wrapper.find(NavLink)).toHaveLength(mainMenuItems.length);
  });
});
