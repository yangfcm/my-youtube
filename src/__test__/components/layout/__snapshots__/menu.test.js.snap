// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Menu component should render component 1`] = `
<div>
  <nav
    className="nav nav-tabs nav-fill"
  >
    <NavLink
      activeClassName="active"
      className="nav-item nav-link"
      exact={true}
      key="Recommend"
      to="/"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="thumbs-up"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
         
      Recommend
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-item nav-link"
      exact={false}
      key="Subscriptions"
      to="/channel"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="user-plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
         
      Subscriptions
    </NavLink>
    <NavLink
      activeClassName="active"
      className="nav-item nav-link"
      exact={false}
      key="Play List"
      to="/playlist"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="list"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
         
      Play List
    </NavLink>
  </nav>
</div>
`;
